main/src/ca/uwaterloo/flix/runtime/interpreter/Channel.java

```diff
@@ -0,0 +1,164 @@
+package ca.uwaterloo.flix.runtime.interpreter;
+
+
+import java.util.*;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.locks.*;
+
+public class Channel {
+  private boolean isOpen;
+  private int bufferSize;
+  private LinkedList<Object> queue;
+  private Set<Condition> waitingGetters;
+  private Condition waitingSetters;
+  private Lock lock;
+  private static AtomicInteger GLOBALCOUNTER = new AtomicInteger();
+  private int id;
+
+  public Channel() {
+    this.isOpen = true;
+    this.queue = new LinkedList<>();
+    this.lock = new ReentrantLock();
+    this.waitingGetters = new HashSet<>();
+    this.waitingSetters = lock.newCondition();
+    this.id = GLOBALCOUNTER.getAndIncrement();
+  }
+
```

Comment:
Add more comments, describing each method at a high-level, and each line as appropriate.